transform.go

```diff
@@ -103,11 +103,19 @@ func transformImage(m image.Image, opt Options) image.Image {
 
 	// never resize larger than the original image
 	if !opt.ScaleUp {
-		if w > imgW {
-			w = imgW
+		if w >= imgW {
+			if h == 0 {
```

Comment:
I don't think this will quite work as intended... if both the requested height and width are larger than the original, then this `h == 0` will be false, and `w` won't get set to zero.  Instead, I think it would work (and would read a little easier) to leave the current checks alone, and then add after them:

```go
// if requested width and height match the original, skip resizing
if w == imgW && h == imgH {
    w, h = 0, 0
}
```

In fact, I'd move that outside the `if !opt.ScaleUp` block entirely.  In any case that the requested width and height match the original, then we don't need to do anything.

Not sure how easy it would be to write a test case for this as well, since it's hard to know if the resize methods were called :confused: